x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedUpdateTests.java

```diff
@@ -314,31 +294,30 @@ protected DatafeedUpdate mutateInstance(DatafeedUpdate instance) {
             builder.setIndices(indices);
             break;
         case 5:
-            Map<String, Object> boolQuery = new HashMap<>();
-            if (instance.getQuery() != null) {
-                boolQuery.put("must", instance.getQuery());
+            try {
+                BoolQueryBuilder query = new BoolQueryBuilder();
+                if (instance.getQuery() != null) {
+                    query.must(instance.getParsedQuery(xContentRegistry()));
+                }
+                query.filter(new TermQueryBuilder(randomAlphaOfLengthBetween(1, 10), randomAlphaOfLengthBetween(1, 10)));
+                builder.setQuery(DatafeedConfig.QueryProvider.fromParsedQuery(query));
+            } catch (IOException ex) {
+                fail(ex.getMessage());
```

Comment:
You'll get better error logging if you simply let the `IOException` propagate out of the method - that will log the whole stack trace, not just the message.  The base class signature of `mutateInstance` allows for it to throw an `IOException`, so just add that to the `throws` clause for this method and remove the two `catch (IOException ex)` blocks from this method.